rfcs/2023-04-18-bundle-check.md

```diff
@@ -0,0 +1,124 @@
+---
+Start Date: 18-04-2023
+RFC PR: (leave this empty if no PR yet)
+Blade Issue: (leave this empty if no issue yet)
+---
+
+# Bundle Size Checks <!-- omit in toc -->
+
+### Table Of Contents <!-- omit in toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+- [Detailed Design](#detailed-design)
+    - [Bundle Size Limit \& Master Diff](#bundle-size-limit--master-diff)
+      - [Bundlemon](#bundlemon)
+    - [Tree Shaking Check](#tree-shaking-check)
+- [Drawbacks/Constraints](#drawbacksconstraints)
+- [Alternatives](#alternatives)
+- [Adoption strategy](#adoption-strategy)
+- [How do we educate people?](#how-do-we-educate-people)
+- [Open Questions](#open-questions)
+
+# Summary
+This RFC proposes the integration of bundle checker into Blade to add checks for the bundle size. This will help us to keep track of the bundle size and prevent any regressions in the future.
+
+# Motivation
+As Blade grows, the size of the generated bundles also grows. This can lead to longer load times and increased bandwidth usage for our consumers. To prevent this, we need a way to monitor the size of our generated assets and ensure that they stay within acceptable limits. We also need to ensure that any changes we make to the design system don't inadvertently increase the size of our bundles, which can impact our users' experience.
+
+# Detailed Design
+We want to achieve the following:
+1. Add a check for each PR that ensures size of our bundles don't increase beyond a certain limit.
```

Comment:
1 and 2 seems similar?

Reply:
I'm actually removing 1 because we don't want to add a hard limit to our bundle size right now since it will keep growing. We only want 2 which is to visualize how much impact each PR has on the bundle size